docs/platform-support.md

```diff
@@ -0,0 +1,44 @@
+# Platform Support
```

Comment:
Maybe not on this document/PR, but would also be good to define the process by which a platform can move between tiers (probably needs Collector approver/maintainer sponsorship?)

Reply:
Can we assume that the assignment of a platform to a specific support tier is fully and only dependent on how the platform is tested as per test strategy? Can we further assume that a platform is elevated to an higher support tier once all required tests for this tier are coded, automatically executed and successfully completed? In this cases, we may tie the description for moving to an higher tier together with the test strategy. In-fact, maybe reasonable to separate out the test strategy into another document.

Reply:
I think there are concerns that are not strictly related to testing, e.g. the ones I mention on https://github.com/open-telemetry/opentelemetry-collector/pull/8224#discussion_r1293645840 and https://github.com/open-telemetry/opentelemetry-collector/pull/8224#discussion_r1293642032. Maybe we can tie these to testing, but I am unsure about it (e.g. I don't think with our current release schedule a bug on a well-tested but very niche platform would warrant a bugfix release).